.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,22 @@
+name: TypeScript Unit Tests
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - ready_for_review
+      - synchronize
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - name: Setup Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '18.x'
+      - name: Install Dependencies
+        run: npm install
+      - name: Run Tests
+        run: npm run compile && npm run unit
```

Comment:
Very minor but `npm run test` does this exact thing if you wanted to be more brief

Reply:
Cheers, I actually copied it from the package.json but will do this suggestion

Reply:
That was going to be my suggestion too :)